vault/resource_managed_keys.go

```diff
@@ -0,0 +1,473 @@
+			Required:    true,
+			Description: "The type of key to use",
+		},
+	}
+
+	return setCommonManagedKeysSchema(s)
+}
+
+func getKeyNameFromConfig(d *schema.ResourceData, configType string) string {
+	stateKey := fmt.Sprintf("%s.%d.%s", configType, 0, consts.FieldName)
+
+	return d.Get(stateKey).(string)
+}
+
+func readManagedKeysConfigBlock(d *schema.ResourceData, keyType string, sm schemaMap) (string, map[string]interface{}) {
+	data := map[string]interface{}{}
+	var name string
+
+	blockField := keyType
+	for blockKey := range sm {
+		stateKey := fmt.Sprintf("%s.%d.%s", blockField, 0, blockKey)
+		if v, ok := d.GetOk(stateKey); ok {
+			data[blockKey] = v
+
+			if blockKey == consts.FieldName {
+				name = v.(string)
+			}
+		}
+	}
+
+	return name, data
+}
+
+func getManagedKeysPath(keyType, name string) string {
+	return fmt.Sprintf("sys/managed-keys/%s/%s", keyType, name)
+}
+
+func managedKeysWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	if _, ok := d.GetOk(consts.FieldAWS); ok {
+		awsKeyName, awsData := readManagedKeysConfigBlock(d, consts.FieldAWS, managedKeysAWSConfigSchema())
+		awsKeyPath := getManagedKeysPath(KMSTypeAWS, awsKeyName)
+
+		if _, err := client.Logical().Write(awsKeyPath, awsData); err != nil {
+			return diag.Errorf("error writing managed key %q, err=%s", awsKeyPath, err)
+		}
+	}
+
+	if _, ok := d.GetOk(consts.FieldPKCS); ok {
+		pkcsKeyName, pkcsData := readManagedKeysConfigBlock(d, consts.FieldPKCS, managedKeysPKCSConfigSchema())
+		pkcsKeyPath := getManagedKeysPath(KMSTypePKCS, pkcsKeyName)
+
+		if _, err := client.Logical().Write(pkcsKeyPath, pkcsData); err != nil {
+			return diag.Errorf("error writing managed key %q, err=%s", pkcsKeyPath, err)
+		}
+	}
+
+	if _, ok := d.GetOk(consts.FieldAzure); ok {
+		azureKeyName, azureData := readManagedKeysConfigBlock(d, consts.FieldAzure, managedKeysAzureConfigSchema())
+		azureKeyPath := getManagedKeysPath(KMSTypeAzure, azureKeyName)
+
+		if _, err := client.Logical().Write(azureKeyPath, azureData); err != nil {
+			return diag.Errorf("error writing managed key %q, err=%s", azureKeyPath, err)
+		}
+	}
+
+	// @TODO figure out what the ID should be
+	d.SetId("sys/managed-keys")
+
+	return managedKeysRead(ctx, d, meta)
+}
+
+func managedKeysRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	if _, ok := d.GetOk(consts.FieldAWS); ok {
+		awsKeyName := getKeyNameFromConfig(d, consts.FieldAWS)
+		awsKeyPath := getManagedKeysPath(KMSTypeAWS, awsKeyName)
+		resp, err := client.Logical().Read(awsKeyPath)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		data := map[string]interface{}{}
+		for k := range managedKeysAWSConfigSchema() {
+			if v, ok := resp.Data[k]; ok {
+				data[k] = v
```

Comment:
The fields such as `access_key` and `secret_key` are in fact returned by Vault, but instead they are being returned as `"redacted"`. In this case I don't think using `continue` would help, since we will always be able to get our value using `resp.Data[k]`; if we skip the logic below that specifically handles those fields, then we will be in a perpetual drift state for the `"redacted"` value fields